onnxruntime/core/providers/rocm/RoctracerLogger.h

```diff
@@ -0,0 +1,194 @@
+// (c) Meta Platforms, Inc. and affiliates. Confidential and proprietary.
+
+#pragma once
+
+#include <functional>
+#include <list>
+#include <memory>
+#include <set>
+#include <vector>
+#include <map>
+#include <unordered_map>
+#include <deque>
+#include <atomic>
+
+#include <roctracer.h>
+#include <roctracer_hcc.h>
+#include <roctracer_hip.h>
+#include <roctracer_ext.h>
+#include <roctracer_roctx.h>
+
+
+namespace onnxruntime{
+namespace profiling {
+class RocmProfiler;
+}
+}
+
+class RoctracerActivityBuffer {
+public:
+  // data must be allocated using malloc.
+  // Ownership is transferred to this object.
+  RoctracerActivityBuffer(uint8_t* data, size_t validSize)
+      : data(data), validSize(validSize) {}
+
+  ~RoctracerActivityBuffer() {
+    free(data);
```

Comment:
>data [](http://example.com/codeflow?start=9&length=4)

Should RoctracerActivityBuffer support move/copy? if yes should we reset data? Otherwise we should mark then as deleted explicitly.

Reply:
I don't anticipate anyone reusing this structure, it is just a wrapper to make sure some malloc memory is freed.  These will never be copied or moved.  Just keeping it simple, no interest in copy semantics or blocking operator= or copy constructors.

Reply:
Then u should mark move/copy as deleted.